src/components/input/index.jsx

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useState, useRef } from "react";
+
+import PropTypes from "prop-types";
+
+const COMPONENT_CLASS_NAME = "c-input";
+
+export const FIELD_TYPES = {
+	EMAIL: "email",
+	PASSWORD: "password",
+	TEXT: "text",
+};
+
+const INPUT_STATUS = {
+	DEFAULT: "default",
+	ERROR: "error",
+	SUCCESS: "success",
+	WARNING: "warning",
+};
+
+const getDerivedInputState = ({ valid, inputState }) => {
+	// regex input validation overrides props input state
+	if (!valid) {
+		return INPUT_STATUS.ERROR;
+	}
+
+	// falls back to default if no props input state
+	return inputState || INPUT_STATUS.DEFAULT;
+};
+
+const Input = ({
+	autoComplete,
+	className,
+	defaultValue = "",
+	hidden = false,
```

Comment:
We should be consistent and use `defaultProps` instead of setting them inline

```
Input.defaultProps = { }
```

Reply:
ok will do

Reply:
addressed